dapps/v2Explorer/src/components/ExplorerModal.tsx

```diff
@@ -0,0 +1,165 @@
+import React, {useCallback, useEffect, useState} from 'react';
+import {StyleSheet, View, useColorScheme, Dimensions} from 'react-native';
+import Modal from 'react-native-modal';
+import {InitialExplorerContent} from './InitialExplorerContent';
+import {ViewAllExplorerContent} from './ViewAllExplorerContent';
+
+//@ts-expect-error - `@env` is a virtualised module via Babel config.
+import {fetchInitialWallets, fetchViewAllWallets} from '../utils/ExplorerUtils';
+import {ExplorerModalHeader} from './ExplorerModalHeader';
+
+const MODAL_HEIGHT = Dimensions.get('window').height * 0.7;
+const DEVICE_WIDTH = Dimensions.get('window').width;
+
+interface ExplorerModalProps {
+  modalVisible: boolean;
+  close: () => void;
+}
+
+// Populate with the data...
+export function ExplorerModal({modalVisible, close}: ExplorerModalProps) {
+  const [isLoading, setIsLoading] = useState(true);
+  const [isViewAllLoading, setViewAllLoading] = useState(true);
+
+  const [viewAllContentVisible, setViewAllContentVisible] = useState(false);
+  const [explorerData, setExplorerData] = useState([]);
+  const [viewAllExplorerData, setViewAllExplorerData] = useState([]);
+
+  const isDarkMode = useColorScheme() === 'dark';
+
+  const fetchWallets = useCallback(() => {
```

Comment:
@bkrem my refactoring as per suggestion.

Reply:
Much cleaner, thank you 🙏